generators/app/templates/src/containers/TicTacToe/GamePage/children/Game/game.component.js

```diff
@@ -0,0 +1,415 @@
+            label: t('noInboxOpponent.link.label'),
+            href: t('noInboxOpponent.link.href')
+          });
+        /**
+         * Find opponent game inbox if doesn't exist get global
+         * @to: Opponent inbox path
+         */
+        const to = notification.getDefaultInbox(inboxes, 'Game', 'Global');
+        /**
+         * Send notification
+         */
+        if (to) {
+          await createNotification(content, to.path);
+        }
+      } catch (error) {
+        errorToaster(error.message, 'Error');
+      }
+    },
+    [gameData, inboxUrl]
+  );
+
+  const getSecondToken = useCallback(token => (token === 'X' ? 'O' : 'X'));
+
+  const generateMoves = useCallback((moveorder: Array<String>, firstmove: String) =>
+    moveorder.reduce((allSquares, current, i) => {
+      const squares = allSquares;
+      let move;
+      if (i % 2 === 0) move = firstmove;
+      else move = getSecondToken(firstmove);
+      squares[current] = move;
+      return squares;
+    }, new Array(9).fill(null))
+  );
+
+  const getPredicate = useCallback(field => {
+    const prefix = tictactoeShape['@context'][field.prefix];
+    return `${prefix}${field.predicate}`;
+  });
+
+  const canPlay = useCallback(({ gamestatus, token }) => {
+    const isStatusValid = gamestatus && gamestatus.includes('Move');
+    return isStatusValid ? gamestatus.includes(token) : false;
+  });
+
+  const changeGameStatus = useCallback(
+    async gamestatus => {
+      try {
+        const predicate = 'http://www.w3.org/2000/01/rdf-schema#gamestatus';
+        await gameDocument[predicate].set(gamestatus);
+      } catch (e) {
+        throw e;
+      }
+    },
+    [gameDocument]
+  );
+
+  const getPlayerInfo = async webId => {
+    try {
+      const nameData = await ldflex[webId]['vcard:fn'];
+      const imageData = await ldflex[webId]['vcard:hasPhoto'];
+      const name = nameData ? nameData.value : webId;
+      const image = imageData ? imageData.value : '/img/icon/empty-profile.svg';
+      return { name, image, webId };
+    } catch (e) {
+      throw e;
+    }
+  };
+
+  const addGameToList = async () => {
+    const url = buildPathFromWebId(webId, process.env.REACT_APP_TICTAC_PATH);
+    await ldflex[`${url}/data.ttl`]['ldp:contains'].add(namedNode(gameURL));
+  };
+
+  const onAccept = async cb => {
+    try {
+      await changeGameStatus('Move X');
+      await addGameToList();
+      await sendNotification(rival.webId, {
+        title: 'Tictactoe game accepted',
+        summary: 'has accepted your invitation to play a game of TicTacToe.',
+        actor: webId,
+        object: gameURL,
+        target: window.location.href
+      });
+      cb();
+    } catch (e) {
+      setIsProcessing(false);
+      errorToaster(e.message, 'Error');
+    }
+  };
+
+  const onDecline = async cb => {
+    try {
+      await changeGameStatus('Declined');
+      await sendNotification(rival.webId, {
+        title: 'Tictactoe game declined',
```

Comment:
Should these be internationalized? Hmm.. will have to think about how to handle i18n notifications

Reply:
This we'll have to discuss. I will resolve it for now